flight/Modules/Actuator/actuator.c

```diff
@@ -338,7 +338,7 @@ static void actuator_task(void* parameters)
 float process_mixer(const int index, const float curve1, const float curve2,
 		ActuatorDesiredData *desired)
 {
-	int8_t *vector = get_mixer_vec(index);
+	typeof(&(*mixerSettings.Mixer1Vector)) vector = get_mixer_vec(index);
```

Comment:
Here's an example:

https://gist.github.com/mlyle/2fc29501dd7ad15ba2a6

During compilation it prints out the warnings you'd want:

```
jar% gcc -o example example.c -O2 -Wall
example.c: In function ‘myFunc2’:
example.c:13:2: warning: return from incompatible pointer type [enabled by default]
  return &baaz;
  ^
example.c: In function ‘main’:
example.c:27:26: warning: array subscript is above array bounds [-Warray-bounds]
  printf("%d\n", (*vector)[16]);
```